### ISSUE TITLE
[analyzer] Type aliases are not working well with the `always_specify_types` rule 

### ISSUE THREAD

**Issue by AlexV525** [dart-lang/linter#2692]
```dart
typedef JsonMap = Map<String, dynamic>;
typedef JsonList = List<JsonMap>;
```
![image](https://user-images.githubusercontent.com/15884415/120604010-2ffe0780-c47f-11eb-87a6-04f5b24e811f.png)

This tracker is for issues related to: Analyzer

* Dart SDK Version: 2.13.1
* macOS BigSur

**Comment by AlexV525** [comment]
@srawlins Any priority on this? It's preventing us to use type alias since it'll raises warnings.

**Comment by srawlins** [comment]
@AlexV525 no priority on this yet.

@pq I'm not sure how we prioritize false positives...

**Comment by bwilkerson** [comment]
In my opinion, false positives should generally have a high priority because they erode user trust.

**Comment by AlexV525** [comment]
Another false positive about `always_specify_types` #2671 .

**Comment by pq** [comment]
+1 to what Brian said.  As far as prioritizing relative to other false positives, I (informally) apply a multiplier for issues impacting core/recommended/pedantic/flutter lints.  I'm looking at making this more formal FWIW...  (And if anyone wants to brainstorm about that, let me know!)

**Comment by srawlins** [comment]
I think each false positive would get a priority bump also because they don't require cleanup, if the fix really means that no new cases would be reported.

**Comment by AlexV525** [comment]
And now it's the "New feature has come, but it's not really come" thing since we can't really put it into production... 😓 An issue for a new feature might have a priority bump too?

**Comment by srawlins** [comment]
Can you not `// ignore` the lint reported? I don't think this should block adoption of new type aliases.

**Comment by AlexV525** [comment]
We could, but we shouldn't. The profit that type aliases bring is less than the code quality drop's effects. Consider that we have multiple classes that would need to use `Json` alias, then we'll going to ignore it everywhere when we use it.

**Comment by AlexV525** [comment]
And after the brainstorm here this issue still don't have a priority which is still blocking us to migrate...
